packages/services/src/kernel/future.ts

```diff
@@ -19,10 +19,11 @@ declare var setImmediate: any;
  * is considered done when the `idle` status is received.
  *
  */
-export class KernelFutureHandler<
-  REQUEST extends KernelMessage.IShellMessage = KernelMessage.IShellMessage,
-  REPLY extends KernelMessage.IShellMessage = KernelMessage.IShellMessage
+export abstract class KernelFutureHandler<
+  REQUEST extends KernelMessage.IShellControlMessage,
+  REPLY extends KernelMessage.IShellControlMessage
 > extends DisposableDelegate implements Kernel.IFuture<REQUEST, REPLY> {
+  abstract async handleMsg(msg: KernelMessage.IMessage): Promise<void>;
```

Comment:
Do you think it would be cleaner for handleMsg to have a default handling of stdin and iopub messages, and subclasses to call the superclass handleMsg?

Reply:
Indeed that would prevent code duplication